### PR TITLE
Prevent admins from deleting themselves

### LINKED ISSUE #37: Prevent admins from being able to delete themselves...

### PR DESCRIPTION
**Pull request by jordanmaguire**
Fix #37

### CODE REVIEW

File: app/views/admin/users/index.html.haml [hunk 1]
```diff
@@ -17,7 +17,9 @@
       %tr
         %td= user.email
         %td.actions
-          = link_to "Edit", edit_admin_user_path(user), class: "btn btn-small"
-          = link_to "Delete", admin_user_path(user), method: :delete, data: { confirm: "Are you sure you want to delete #{user}?" }, class: "btn btn-small"
+          - if policy(user).edit?
```
Comment (pzi) [inline app/views/admin/users/index.html.haml]: OOC, how'd you feel about having disabled buttons instead?

Comment (jordanmaguire) [inline app/views/admin/users/index.html.haml]: Prefer not - what benefits do you see? I think part of not being able to do something is not knowing that the something is even an option.

Comment (pzi) [inline app/views/admin/users/index.html.haml]: Just a thought...

People are smart; there is a column in that table do to something even if the buttons in it are missing.

Maybe visually more appealing if all the options are listed but you cannot use them.


Randomly thought of analogy:
If a person is not meant to access the room behind a door, you don't hide it, you lock it. :wine_glass: :smoking:

**Comment by swebb** [comment]
:+1:

File: app/views/admin/users/index.html.haml (hunk 1 above)
Comment (psyren89) [inline app/views/admin/users/index.html.haml]: I'd lean towards not having the button shown at all. The less things people have to play with, the less they have to complain about/break.

File: app/policies/user_policy.rb [hunk 2]
```diff
@@ -29,6 +29,11 @@ def member_dashboard?
 
 # CRUD
 
+  # Admins can't delete themselves.
+  def destroy?
+    is_admin? && user != record
```
Comment (psyren89) [inline app/policies/user_policy.rb]: I might be reading this wrong, but what exactly is the condition here?

Comment (jordanmaguire) [inline app/policies/user_policy.rb]: the method answers the question: "Can the currently signed in user (user) delete the given user? (record)"

The answer is "Yes: If the user is an admin and the user it is trying to delete (record) is not themselves (user)"

Comment (psyren89) [inline app/policies/user_policy.rb]: :+1:

**Pull request merged**